charms/knative-serving/src/lightkube_helpers/__init__.py

```diff
@@ -0,0 +1,88 @@
+from typing import List, Union, Tuple
+
+import lightkube
+from lightkube.core.resource import NamespacedResource, GlobalResource
+from lightkube.resources.apps_v1 import StatefulSet
+from ops.model import ActiveStatus, WaitingStatus, BlockedStatus, MaintenanceStatus
+
+from charm_helpers.exceptions import ErrorWithStatus
+
+
+def check_resources(client: lightkube.Client, expected_resources: List[Union[NamespacedResource, GlobalResource]]) -> \
+        (bool, List[Exception]):
+    """Checks status of resources in cluster, returning True if all are considered ready
+
+    Note: This is a basic skeleton of a true check on the resources.  Currently it only checks
+    that resources exist and that StatefulSets have their desired number of replicas ready.
+
+    Returns: Tuple of:
+        Status (bool)
+        List of Exceptions encountered during failed checks, with each entry
+        indexed the same as the corresponding expected_resource (list[str])
+    """
+    errors = [None] * len(expected_resources)
+    for i, expected_resource in enumerate(expected_resources):
+        try:
+            found_resource = client.get(
+                type(expected_resource),
+                expected_resource.metadata.name,
+                namespace=expected_resource.metadata.namespace,
+            )
+        except lightkube.core.exceptions.ApiError as e:
```

Comment:
Is an `ApiError` always going to mean the resource was not found? Should we maybe check the error message before assuming this?

Reply:
Good question - I think you're right that it does not.  I believe there's a 404 return code I can check for - will update that shortly